experiments/misc/ilamb_conversion.jl

```diff
@@ -0,0 +1,389 @@
+        ),
+    )
+    NCDatasets.defVar(
+        ilamb_ds,
+        "time_bnds",
+        # Even though the dates need to be shifted by one month, the
+        # date/time bounds are correct
+        Array(clima_ds["date_bnds"]),
+        ("bnds", "time"),
+        attrib = Dict("units" => "days since 1850-01-01"),
+    )
+    return nothing
+end
+
+"""
+    define_var(ilamb_ds, clima_ds, clima_short_name)
+
+Define the variable with the name `clima_short_name` in `ilamb_ds` using
+`clima_ds`.
+"""
+function define_var(ilamb_ds, clima_ds, clima_short_name)
+    # Use Float32 instead of Float64 to save space
+    FT = Float32
+    ilamb_mapping = ILAMB_VARIABLES[clima_short_name]
+    (; short_name, long_name, conversion_factor, units, clima_units) =
+        ilamb_mapping
+    var_attribs = Dict(
+        "_FillValue" => FT(1.0e20),
+        "standard_name" => short_name,
+        "long_name" => long_name,
+        "units" => units,
+    )
+    # For ILAMB, the order of dimensions should be lon, lat, and time
+    expected_dims = ["lon", "lat", "time"]
+    data = get_data(
+        clima_ds,
+        clima_short_name,
+        clima_units,
+        conversion_factor,
+        expected_dims,
+        FT,
+    )
+    NCDatasets.defVar(
+        ilamb_ds,
+        short_name,
+        data,
+        Tuple(expected_dims),
+        attrib = var_attribs,
+    )
+    return nothing
+end
+
+"""
+    get_data(
+        clima_ds,
+        clima_short_name,
+        clima_units,
+        conversion_factor,
+        expected_dims
+    )
+
+Get the data from `clima_ds` with the name `clima_short_name`.
+
+Additional preprocessing includes converting to units that ILAMB expects with
+`conversion_factor` and permuting the dimensions with `expected_dims`.
+"""
+function get_data(
+    clima_ds,
+    clima_short_name,
+    clima_units,
+    conversion_factor,
+    expected_dims,
+    FT,
+)
+    data = Array(clima_ds[clima_short_name])
+
+    # Special cases for tsoil and swc to extract top 10 cm (z >= -0.1)
+    if clima_short_name in ("tsoil", "swc")
+        z = Array(clima_ds["z"])
+        all_dimnames = NCDatasets.dimnames(clima_ds[clima_short_name])
+        top_layer_indices = findall(val -> val >= -0.1, z)
+        length(top_layer_indices) >= 1 ||
+            error("No valid depths above -0.1m found for $(clima_short_name)")
+        z_idx = last(top_layer_indices)
+
+        index_tuple =
+            (dimname == "z" ? z_idx : Colon() for dimname in all_dimnames)
+        data = data[index_tuple...]
+    end
+
+    # Permute dimensions to match ILAMB
+    dimnames = filter(
+        dimname -> dimname in expected_dims,
+        NCDatasets.dimnames(clima_ds[clima_short_name]),
+    )
+    perm = indexin(expected_dims, collect(dimnames))
+    clima_ds[clima_short_name].attrib["units"] != clima_units && error(
+        "Units are not the same for $clima_short_name. Update the conversion factor and units in the mapping and run this script again",
```

Comment:
I think this can be more descriptive since there are a lot of units going around, something like:

```suggestion
        "Units in the simulation-output NetCDF do not match the expected CliMA units for $clima_short_name. Update the conversion factor and units in the mapping and run this script again",
```